### PR TITLE
Refuse to accept 2nd connection of already connected taker

### PR DESCRIPTION
**Pull request by thomaseizinger**
Noticed that we currently don't check for existing connections of the same taker.

Introduce a utility on `Tasks` to make it easier to run fallible futures in the background.
I've split this out of https://github.com/itchysats/itchysats/pull/851 which is not quite done yet in the hope that these two things are pretty non-controversial!

### CODE REVIEW

File: daemon/src/lib.rs
Comment (luckysori) [inline daemon/src/lib.rs:121]: 🔧 `if let Err(e)` might be ever so slightly cleaner.

**Review by luckysori — APPROVED** [review]

File: daemon/src/lib.rs
Comment (thomaseizinger) [reply daemon/src/lib.rs:121]: I wanted to be explicit that the `Ok` part returns `()` but maybe it is not worth it.

**Comment by thomaseizinger** [comment]
Feel free to merge once it has some approvals.

**Review by klochowicz — APPROVED** [review]
LGTM!

**Pull request merged**
